Frameworks/Foundation/NSKVOSupport.mm

```diff
@@ -238,7 +264,7 @@ static void _addNestedObserversAndOptionallyDependents(_NSKVOKeyObserver* keyObs
 
     // Aggregate all keys whose values will affect us.
     if (dependents) {
-        NSSet* valueInfluencingKeys = [[object class] keyPathsForValuesAffectingValueForKey:key];
+        NSSet* valueInfluencingKeys = _keyPathsForValuesAffectingValueForKey([object class], key);
```

Comment:
>valueInfluencingKeys [](start = 15, length = 20)

could be nil. messages to nil object should be okay but i would double check this on ARM (we have had funky issues there but not with integral type rets).